python-backend/app/api/mines/notification/models/notification.py

```diff
@@ -0,0 +1,32 @@
+import uuid
+from app.api.utils.include.user_info import User
+from sqlalchemy.schema import FetchedValue
+from sqlalchemy.dialects.postgresql import UUID
+from app.extensions import db
+from app.api.utils.models_mixins import  Base
+
+
+class Notification( Base):
+    __tablename__ = 'notification'
+    user_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    mine_guid = db.Column(UUID(as_uuid=True),db.ForeignKey('mine.mine_guid'))
+    idir = db.Column(db.String, nullable=False)
+
+    @classmethod
+    def create_subscription(cls, mine_guid):
+        user_name = User().get_user_username()
+        if Notification.query.filter_by(mine_guid=uuid.UUID(mine_guid)).filter_by(
+            idir=user_name).first() is None:
+            subscription = cls(mine_guid=uuid.UUID(mine_guid), idir=user_name)
+            subscription.save()
+            return subscription
+        return
+
+    @classmethod
+    def delete_subscription(cls, mine_guid):
+        user_name = User().get_user_username()
+        subscription_to_delete = Notification.query.filter_by(mine_guid=uuid.UUID(mine_guid)).filter_by(idir=user_name).first()
```

Comment:
This query should be split into a `find_by_mine_guid` class method. And then wherever it's used, it should have safety around it to make sure that the `subscription_to_delete` is not `None`. If it is, a 404 or something similar should be thrown to say that the subscription was not found.

The same comment applies to the use of this query in `create_subscription`.